nutanix/resource_nutanix_ndb_database_snapshot.go

```diff
@@ -0,0 +1,587 @@
+							Computed: true,
+						},
+						"async": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"stand_by": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"curation_retry_count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"operations_using_snapshot": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"software_snapshot_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"software_database_snapshot": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"dbserver_storage_metadata_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"santised": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"santised_from_snapshot_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"timezone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"processed": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"database_snapshot": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"from_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"to_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dbserver_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dbserver_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dbserver_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"replicated_snapshots": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"software_snapshot": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"santised_snapshots": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"snapshot_family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"snapshot_timestamp_date": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"lcm_config": dataSourceEraLCMConfig(),
+			"parent_snapshot": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"snapshot_size": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceNutanixNDBDatabaseSnapshotCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	req := &era.DatabaseSnapshotRequest{}
+	snapshotName := ""
+	tmsID, tok := d.GetOk("time_machine_id")
+	tmsName, tnOk := d.GetOk("time_machine_name")
+
+	if !tok && !tnOk {
+		return diag.Errorf("Atleast one of time_machine_id or time_machine_name is required to perform clone")
+	}
+
+	if len(tmsName.(string)) > 0 {
+		// call time machine API with value-type name
+		res, err := conn.Service.GetTimeMachine(ctx, "", tmsName.(string))
```

Comment:
Instead of using "" using parameter="" for readability

Reply:
done.